types/mongoose/mongoose-tests.ts

```diff
@@ -1213,8 +1213,13 @@ MongoModel.find({}).$where('indexOf("val") !== -1').exec(function (err, docs) {
   docs[0].__v;
 });
 MongoModel.findById(999, function (err, doc) {
+  var handleSave = function(err: Error, product: mongoose.Document, numAffected: number) {};
   doc.increment();
-  doc.save(cb).then(cb).catch(cb);
+  doc.save(handleSave).then(cb).catch(cb);
```

Comment:
You should add these 2 test cases still:
```js
doc.save(function(err, product, numAffected) {}).then(cb).catch(cb);
doc.save({
  validateBeforeSave: false
}, function(err, product, numAffected) {}).then(cb).catch(cb);
```
The reason is that we want to test if Typescript can infer the types of `(err, product, numAffected)` instead of manually defining types above.